### PR TITLE
Ref/motor queries

### PR DESCRIPTION
**Pull request by joshLong145**
refactoring of `Query` implementation to be explicit calls on `motor` interface. This is a place holder until we can abstract querying implementations to a generic endpoint on the `motor` interface capable of handling all requirements of querying.
## Changes
- refactored `query.go` to hold query logic, but to no longer go through the `Query` method
- added all querying methods to `motor` interface for explicit interfaction.
## API Updates
the following methods have been added to motor 
```
QueryBucket(req mt.QueryWhatIsRequest) (*mt.QueryWhereIsResponse, error)
QueryBucketGroup(req mt.QueryWhereIsByCreatorRequest) (*mt.QueryWhereIsByCreatorResponse, error)
QueryRegistry(req mt.QueryWhoIsRequest) (*mt.QueryWhoIsResponse, error)
QuerySchema(req mt.QueryWhatIsRequest) (*mt.QueryWhatIsResponse, error)
```
### Deprecations *(required)*
- deprecates `Query` method in favor of individual query methods.

## Checklist

- [ ] Unit tests
- [ ] Documentation

## References *(optional)*

Fixes 
Connects 

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

**Review by prdlk — APPROVED** [review]
LGTM

**Review by 6missedcalls — APPROVED** [review]
LGTM

File: pkg/motor/query.go [hunk 1]
```diff
@@ -82,34 +38,33 @@ func (mtr *motorNodeImpl) handleBucketGroupQuery(addr string) (*mt.QueryResultIt
...
-		WhereIsList: ptrArr,
+	return &mt.QueryWhereIsByCreatorResponse{
+		Code:    http.StatusAccepted,
+		WhereIs: ptrArr,
 	}, nil
 }
 
-func (mtr *motorNodeImpl) handleRegistryQuery(did string) (*mt.QueryResultItem, error) {
-	resp, err := mtr.GetClient().QueryWhoIs(did)
+func (mtr *motorNodeImpl) QueryRegistry(req mt.QueryWhoIsRequest) (*mt.QueryWhoIsResponse, error) {
```
Comment (mcjcloud) [inline pkg/motor/query.go:47]: nit: name this `QueryWhoIs` as it's more accurate, and more things could potentially be added to registry in the future creating ambiguity.

**Review by mcjcloud — APPROVED** [review]
Just one comment

File: pkg/motor/query.go (hunk 1 above)
Comment (joshLong145) [reply pkg/motor/query.go:47]: i would agree, but at this time its the only thing we are querying from the registry module. to keep it consistent i went with that name.

**Pull request merged**